aura.grammar

```diff
@@ -0,0 +1,56 @@
+
+use_statement: identifier ['as' identifier]
+
+function_def: 'fn' NAME '(' decl_list ')' type NEWLINE function_body
+
+return_statement: 'ret' [expression]
+
+decl_list: decl | decl ',' decl_list | epsilon
+
+type: i[1-9][0-9]* | f16 | f32 | f64 | f128 | '[' type ']' | '{' type ':' type '}'
+
+
+array : '[' identifier_list ']'
+
+assignment_statement: identifier type '=' expression
+                    | '!' assignment_statement
+
+if_statement: 'if' expression NEWLINE body
```

Comment:
Should we have `NEWLINE` here? For instance, `trait` doesn't have it but still requires a newline for the `function_decls` rule. I think since it's a whitespace-based language, it is somewhat implied there is a line break, but at the same time it should be specified that there are line-breaks for such constructs. What do you think?